packages/evm/src/memory.ts

```diff
@@ -30,7 +32,10 @@ export class Memory {
     const newSize = ceil(offset + size, 32)
     const sizeDiff = newSize - this._store.length
     if (sizeDiff > 0) {
-      this._store = Buffer.concat([this._store, Buffer.alloc(sizeDiff)])
+      this._store = Buffer.concat([
+        this._store,
+        Buffer.alloc(Math.ceil(sizeDiff / CONTAINER_SIZE) * CONTAINER_SIZE),
+      ])
```

Comment:
Ok, just did a quick search `._store.length` for the EVM to see if there are any external accesses to memory length where we would have side effects, but this seems not to be the case (apart from some memory-internal tests which can eventually be adopted if needed). 🙂

So this seems like a really clever solution to reduce the buffer allocations drastically along memory extensions! 👍

(still am a bit cautious and wonder though if this might have some side effect(s) *somewhere*? 🤔

Reply:
I agree that we should very carefully examine this. (I also did the same search as you did since I was afraid that there indeed might be external reads to it)